pkg/query/columnquery.go

```diff
@@ -918,51 +933,21 @@ func sliceRecord(r arrow.Record, indices []int64) []arrow.Record {
 	return slices
 }
 
-func (q *ColumnQueryAPI) getMappingFiles(
+func getMappingFilesAndLabels(
 	ctx context.Context,
-	m *pb.MergeProfile,
-) ([]string, error) {
-	start := time.Time{}
-	if m.Start != nil {
-		start = m.Start.AsTime()
-	}
-
-	end := time.Time{}
-	if m.End != nil {
-		end = m.End.AsTime()
-	}
-
-	p, err := q.querier.GetProfileMetadataMappings(
-		ctx,
-		m.Query,
-		start,
-		end,
-	)
+	q Querier,
+	query string,
+	startTime, endTime time.Time,
+) ([]string, []string, error) {
+	mappingFiles, err := q.GetProfileMetadataMappings(ctx, query, startTime, endTime)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return p, nil
-}
-
-func (q *ColumnQueryAPI) getLabels(
-	ctx context.Context,
-	req *pb.LabelsRequest,
-) ([]string, error) {
-	start := time.Time{}
-	if req.Start != nil {
-		start = req.Start.AsTime()
-	}
-
-	end := time.Time{}
-	if req.End != nil {
-		end = req.End.AsTime()
-	}
-
-	l, err := q.querier.GetProfileMetadataLabels(ctx, req.Match, start, end)
+	labels, err := q.GetProfileMetadataLabels(ctx, []string{}, startTime, endTime)
```

Comment:
We should remove the match variable here. We don’t use it and don’t anticipate using it anytime soon.